src/lib/comms/include/sol-connman.h

```diff
@@ -0,0 +1,285 @@
+#define SOL_CONNMAN_SERVICE_TYPE_BLUETOOTH  "bluetooth" /**< @brief bluetooth service type string */
+#define SOL_CONNMAN_SERVICE_TYPE_CELLULAR   "celluar"   /**< @brief bluetooth service type string */
+#define SOL_CONNMAN_SERVICE_TYPE_GPS        "gps"       /**< @brief gps service type string */
+#define SOL_CONNMAN_SERVICE_TYPE_VPN        "vpn"       /**< @brief vpn service type string */
+#define SOL_CONNMAN_SERVICE_TYPE_GADGET     "gadget"    /**< @brief gadget service type string */
+#define SOL_CONNMAN_SERVICE_TYPE_P2P        "p2p"       /**< @brief p2p service type string */
+
+/**
+ * @eium sol_connman_state
+ *
+ * @brief the global connection state of system
+ *
+ * One of these must be chosen for the global connection state,
+ * with sol_connman_get_state()
+ *
+ */
+enum sol_connman_state {
+    SOL_CONNMAN_STATE_UNKNOWN   = 0,
+    SOL_CONNMAN_STATE_IDLE      = 1,
+    SOL_CONNMAN_STATE_READY     = 2,
+    SOL_CONNMAN_STATE_ONLINE    = 3,
+    SOL_CONNMAN_STATE_OFFLINE   = 4,
+};
+
+/**
+ * @brief Gets the service name
+ *
+ * This function gets the name for the connman service,
+ * after the service is to give by service monitor.
+ *
+ * @param service The service structure which the name is desired
+ *
+ * @return The name of the services on success, NULL on failure.
+ */
+const char *sol_connman_service_get_name(
+    const struct sol_connman_service *service);
+
+/**
+ * @brief Gets the service state
+ *
+ * This function gets the state for the connman service,
+ * after the service is to give by service monitor.
+ *
+ * @note SOL_CONNMAN_SERVICE_STATE_REMOVE is used to show
+ * the service has been removed. When it is notified,
+ * the service has been removed in the system.
+ *
+ * @param service The service structure which the state is desired
+ *
+ * @return The state of the services on success,
+ * SOL_CONNMAN_SERVICE_STATE_UNKNOWN on failure.
+ */
+enum sol_connman_service_state sol_connman_service_get_state(
+    const struct sol_connman_service *service);
+
+/**
+ * @brief Gets the service type
+ *
+ * This function gets the type for the connman service,
+ * after the service is to give by service monitor.
+ *
+ * @param service The service structure which the type is desired
+ *
+ * @return The type of the services on success,
+ * SOL_CONNMAN_SERVICE_TYPE_UNKNOWN on failure.
+ */
+const char *sol_connman_service_get_type(
+    const struct sol_connman_service *service);
+
+/**
+ * @brief Gets the service network address
+ *
+ * This function gets the network address for the connman service,
+ * after the service is to give by service monitor.
+ *
+ * @param service The service structure which the network address is desired
+ * @param family which type of the network address is desired (ipv4 or ipv6).
+ *
+ * @return A point to sol_network_link_addr created on success, NULL on failure.
+ */
+struct sol_network_link_addr *sol_connman_service_get_network_address(
```

Comment:
In sol-network.h we have `struct sol_network_link`, that contains a list of addresses and a flag, both look meaningful here, so maybe get that? Also the pattern could be to fill a given pointer instead of allocating and returning it, then return -errno.